frontend/src/components/pages/Dropoff/index.tsx

```diff
@@ -0,0 +1,80 @@
+import { Button, Container, Text } from "@chakra-ui/react";
+import React, { useContext, useState } from "react";
+import { useHistory } from "react-router-dom";
+
+import DonorAPIClient from "../../../APIClients/DonorAPIClient";
+import SchedulingAPIClient from "../../../APIClients/SchedulingAPIClient";
+import * as Routes from "../../../constants/Routes";
+import AuthContext from "../../../contexts/AuthContext";
+import { DonorResponse } from "../../../types/DonorTypes";
+import { Schedule } from "../../../types/SchedulingTypes";
+import DropoffCard from "./components/DropoffCard";
+
+const Dropoff = (): JSX.Element => {
+  const { authenticatedUser } = useContext(AuthContext);
+  const [schedule, setSchedule] = useState<Schedule[] | null>([]);
+  const history = useHistory();
+
+  if (!authenticatedUser) {
+    history.push(Routes.LOGIN_PAGE);
+  }
+
+  const deleteSchedule = async (id: string) => {
+    await SchedulingAPIClient.deleteSchedule(id);
+    if (schedule) {
+      setSchedule(schedule.filter((s) => s?.id !== id));
+    }
+  };
+
+  React.useEffect(() => {
+    const getSchedules = async () => {
+      const donorID =
+        (await DonorAPIClient.getAllDonors()).find(
```

Comment:
I really think we should not be getting all donors and then filtering. We can have `getDonorByUserID` in backend which does this work. I can make this change in a new PR and get it merged in or if u want you can do it in this same PR.